### PR TITLE
Make the `target` section optional

### LINKED ISSUE #30: what is ``target`` and coud it made to be optional ?
In our process, and in the documentation, it was not clear at first what purpose would serve the currently mandatory ``target`` option in yaml config file. After doing some tinkering, it seems to me that it actually should be optional, it could maybe be required only for special occasion (thinking of push functionality)... but even then, it seems it would be better moved away.

As we can see in your documentation example, it seems both part of the target the "remote" or the "branch" part are either not used (the remote is only used on push, which is optional), or could be kept as an internal implementation choice or fixed value as it appears to be the branch name where we will merge all merge targets.

I could understand to keep this as an option, but because of its apparent similarity with the format of remotes, which are actually not at all the same, this generates much mis-comprehension. For instance the reference part of ``target`` can only be a branch name (whereas reference part of ``merge`` supports commits sha1 and tags)... and target's branch anyway will be created (or ``reset``ed if existent in the repository) anyway.

For me, to the best of my imperfect knowledge, it seems an improvement to remove this option whenever we don't need it. 

It seems it could be called "push-target", and be optional.

What are your thoughts ?

### PR DESCRIPTION
**Pull request by janverb**
This makes it possible to use the `target` configuration without a remote (in which case pushing will be impossible) and to leave it out completely (in which case a branch called `_git_aggregated` will be used for aggregation).

Resolves #30.

### CODE REVIEW

**Comment by thomaspaulb** [comment]
@sbidoul Is this project still being maintained?

File: tests/test_config.py [hunk 1]
```diff
@@ -252,40 +252,54 @@ def test_load_target_exception(self):
...
+            '"[remote_name] branch_name"')
 
         config_yaml = """
 /product_attribute:
     remotes:
         oca: https://github.com/OCA/product-attribute.git
     merges:
         - oca 8.0
-    target:
+    target: oba aggregated_branch
```
Comment (sbidoul) [inline tests/test_config.py:270]: Can we keep the existing tests and add new ones for the new use cases ?

**Review by sbidoul — COMMENTED** [review]
A comment on the tests.

**Comment by sbidoul** [comment]
> @sbidoul Is this project still being maintained?

Well, not actively apparently. We can certainly welcome additional maintainers.

I'll look at putting CI back to a functioning state, at least.

**Comment by sbidoul** [comment]
Reviewing and testing and looking at open issues would definitely help !

**Comment by sbidoul** [comment]
I've done a bit of maintenance on the CI. Can you please rebase, and keep the existing tests and add new ones ?

File: tests/test_config.py (hunk 1 above)
Comment (janverb) [reply tests/test_config.py:270]: That existing test was testing things you weren't allowed to do:
- Missing `target`
- Empty `target`
- Bad remote in `target`

The first two are allowed now, so I replaced them by an over-full `target` (which is still not allowed). I kept the third and added a test for the new use cases.

Comment (sbidoul) [reply tests/test_config.py:270]: I may misread, but then I don't see the test with Missing target and Empty target.

Comment (janverb) [reply tests/test_config.py:270]: Those are allowed now so the tests had to be removed/replaced. The test tested the error, but they no longer give an error.

I did add a new different test for a missing target. (An empty target is treated the same (`repo_data.get('target', "")`) so I didn't test that separately.)

Comment (sbidoul) [reply tests/test_config.py:270]: > Those are allowed now so the tests had to be removed/replaced.

I don't understand that. If this becomes allowed, the test can stay with the same input, but updated to test the expected outcome.

Comment (janverb) [reply tests/test_config.py:270]: `test_target_defaults` tests the same input with a new expected outcome. It didn't make sense to keep it in a test for testing exceptions, so I moved it.

Because a missing and empty `target` are so similar I only test a missing `target`, but I could add another test for an empty `target` if you prefer.

File: README.rst [hunk 2]
```diff
@@ -181,7 +180,8 @@ Use additional variables from file while expanding:
 
 The env file should contain `VAR=value` lines. Lines starting with # are ignored.
 
-You can also aggregate and automatically push the result to the target:
+You can also aggregate and automatically push the result to the target, if the
+``target`` option is configured:
```
Comment (sbidoul) [inline README.rst:184]: Would it then make sense to also add a documention chunk explaining what happens when target is *not* configured ?

File: tests/test_config.py [hunk 3]
```diff
@@ -252,40 +252,54 @@ def test_load_target_exception(self):
...
+        config_yaml = """
+/product_attribute:
+    remotes:
+        oca: https://github.com/OCA/product-attribute.git
+    merges:
+        - oca 8.0
+"""
+        repos = config.get_repos(self._parse_config(config_yaml))
+        self.assertDictEqual(
+            repos[0]["target"], {"branch": "_git_aggregated", "remote": None}
```
Comment (sbidoul) [inline tests/test_config.py:301]: :+1:

File: README.rst (hunk 2 above)
Comment (sbidoul) [reply README.rst:184]: Yes, it seems important to talk about the `_git_aggregated` default branch in the documentation.

**Comment by sbidoul** [comment]
Looks good, thank! I'll cut a release soon.
